### PR TITLE
Added an Orca wrapper

### PR DESCRIPTION
**Pull request by pultar**
Added class `OrcaCalculator` in the style of the `GaussianCalculator`. 

Implemented features are:

* Singlepoint energy calculations
* Mulliken charges
* Loewdin charges
* Hirshfeld charges
* NMR shielding contants

Features are demonstrates in two new Jupyter notebooks and checked in automated tests.

As of now, the following restrictions apply:

* multiprocessing is not yet implemented
* only version 4.2.0 of Orca has been tested (version 5 has some keywords changed)
* geometry optimizations are not yet possible

Automated tests and Jupyter notebooks work well on my machine.

### CODE REVIEW

File: ppqm/orca.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+        self.health_check()
+
+    def __repr__(self) -> str:
+        return f"OrcaCalc(cmd={self.cmd}, scr={self.scr}, n_cores={self.n_cores}, memory={self.memory}gb)"
+
+    def health_check(self):
+        assert env.which(self.cmd), f"Cannot find {self.cmd}"
+
+        # There is no such thing as "orca --version": https://orcaforum.kofo.mpg.de/viewtopic.php?f=8&t=8181
```
Comment (charnley) [inline ppqm/orca.py:69]: hahaha

**Comment by pultar** [comment]
Quick summary and to do list of your comments:

- [ ] remove test.md
- [ ] add partial success for read_properties
- [ ] Whenever you read a file, checking for keywords, assume that it won't be found
- [ ] Consider making the multi-processing interface
- [ ] make sure tests for NH4+ consistently give the same geometries and energies
- [ ] run pre-commit
- [ ] adjust linesio function for blocks without explicit end statements

**Comment by pultar** [comment]
I merged `master` into `orca-wrapper` and added tests for an orca 5.0.2 output file. The parsers extract all supported properties successfully. I also ran the script on our cluster and it worked well. A side note for the tests: I found that `rdkit` generates slightly different conformers depending on the version used. Calculated SP energies might thus be slightly off and the test may fail. I used version `2020.09.3` on both Linux and Mac.

- [x] support for other orca versions
- [x] tests for orca 5.0.2 output files
- [x] supported plattforms for tests: Mac / Linux 
- [x] more property parsers
- [x] Gibbs free energy, enthalpy, entropy, vibrational frequencies, position on the PSE (minimum, TS, higher order)
- [x] multiprocessing interface

**Comment by charnley** [comment]
Looks great! I will do the last review tomorrow and merge

**Pull request merged**